plugins/chat/plugin.rb

```diff
@@ -381,6 +382,41 @@ module ::Chat
     end
   end
 
+  # outgoing webhook events
+  %i[
+    chat_message_created
+    chat_message_edited
+    chat_message_trashed
+    chat_message_restored
+  ].each do |chat_message_event|
+    on(chat_message_event) do |message, channel, user|
+      guardian = Guardian.new(user)
+      memberships = Chat::ChannelMembershipManager.all_for_user(user)
+
+      payload = {
+        message: Chat::MessageSerializer.new(message, { scope: guardian, root: false }).as_json,
+        channel:
+          Chat::ChannelSerializer.new(
+            channel,
+            {
+              scope: guardian,
+              membership:
+                memberships.find { |membership| membership.chat_channel_id == channel.id },
+              root: false,
+            },
+          ).as_json,
+      }
+
+      category_id = channel.chatable_type == "Category" ? channel.chatable_id : nil
+
+      WebHook.enqueue_chat_message_hooks(
+        chat_message_event,
+        payload.to_json,
+        category_id: category_id,
```

Comment:
We will already have this info in channel["chatable"] of the result

Reply:
This isn't for the serialization. 

It is for the webhook emitter to be able to filter the webhook based on the category associated with the channel.

I'm adding it here because I'm already serializing the data to JSON at this point